### ISSUE TITLE
About Demo_RSSS loss issue

### ISSUE THREAD

**Issue by holmeslong** [Cwuwhu/FCD-GAN-pytorch#1]
Thank you for your outstanding contribution! I would like to ask when I train the network with Demo_RSSS, I find that the segmentor loss is increasing while the discriminator loss keeps decreasing, do you have any suggestions when I encounter this situation?

**Comment by Cwuwhu** [comment]
Thank you for pay your attention to our work. Mostly, discriminator is much easier to converge, and segmentor is more difficult to train. So, when the discriminator is much powerful, the balance between adversarial process will be broken. So, when I face this problem, I will try to decrease the learning rate of the optimizer of discriminator, or increase the lr of segmentor. You can have a try.



***@***.***
 
From: holmeslong
Date: 2023-02-27 16:41
To: Cwuwhu/FCD-GAN-pytorch
CC: Subscribed
Subject: [Cwuwhu/FCD-GAN-pytorch] About Demo_RSSS loss issue (Issue #1)
Thank you for your outstanding contribution! I would like to ask when I train the network with Demo_RSSS, I find that the segmentor loss is increasing while the discriminator loss keeps decreasing, do you have any suggestions when I encounter this situation?

**Comment by holmeslong** [comment]
Thank you for your reply, it helps me a lot. While reading the code in the Demo_RSSS section, I came across this line of code. Although the default setting of discriminator_continuous is True, the sign function causes a gradient of 0 on backward propagation when discriminator_continuous is False. What should I do about this problem?
![image](https://user-images.githubusercontent.com/56954102/226841719-fef9b1c3-d6b9-42de-ab5a-89e9599acd5b.png)

**Comment by Cwuwhu** [comment]
Mostly, the soft mask can obtain a good performance. I remain the hard mask, since it may work in some special situations. I hope it also provide some new trials for readers. I will check if the hard mask will have some problems in this code. Thank you for your notice.

**Comment by holmeslong** [comment]
OK, I understand. Thank you for your reply.
